src/TorchVision/AutoAugment.cs

```diff
@@ -0,0 +1,197 @@
+                        return F.solarize(img, magnitude);
+                    case opType.AutoContrast:
+                        return F.autocontrast(img);
+                    case opType.Equalize:
+                        return F.equalize(img);
+                    case opType.Invert:
+                        return F.invert(img);
+                    case opType.Identity:
+                        return img; // Pass
+                    default:
+                        throw new ArgumentException($"The provided operator {op_name} is not recognized.");
+                }
+            }
+        }
+
+        /* Original implementation from:
+         * https://pytorch.org/vision/main/_modules/torchvision/transforms/autoaugment.html#RandAugment */
+        internal class RandAugment : AutoAugmentBase, ITransform
+        {
+            public RandAugment(
+                int num_ops = 2,
+                int magnitude = 9,
+                int num_magnitude_bins = 31,
+                InterpolationMode interpolation = InterpolationMode.Nearest,
+                IList<float>? fill = null)
+            {
+                this.num_ops = num_ops;
+                this.magnitude = magnitude;
+                this.num_magnitude_bins = num_magnitude_bins;
+                this.interpolation = interpolation;
+                this.fill = fill;
+            }
+
+            public Tensor call(Tensor img)
+            {
+                using var _ = torch.NewDisposeScope();
+                var cDim = img.Dimensions - 3;
+                var height = img.shape[cDim + 1];
+                var width = img.shape[cDim + 2];
+                var op_meta = augmentation_space(num_magnitude_bins, (height, width));
+                for (int i = 0; i < num_ops; ++i) {
+                    var op_index = torch.randint(0, op_meta.Count, 1).ToInt32();
+                    var op_name = op_meta.Keys.ToList()[op_index];
+                    var (magnitudes, signed) = op_meta[op_name];
+                    var magnitude = magnitudes.Dimensions > 0 ? magnitudes[this.magnitude].ToDouble() : 0.0;
+
+                    if (signed && torch.randint(0, 2, 1).ToBoolean())
+                        magnitude *= -1.0;
+
+                    img = apply_op(img, op_name, magnitude, interpolation, this.fill);
+                }
+                return img.MoveToOuterDisposeScope();
+            }
+
+            private Dictionary<opType, (Tensor, bool)> augmentation_space(int num_bins, (long height, long width) image_size)
+            {
+                return new Dictionary<opType, (Tensor, bool)> {
+                    { opType.Identity, (torch.tensor(0.0), false) },
+                    { opType.ShearX, (torch.linspace(0.0, 0.3, num_bins), true) },
+                    { opType.ShearY, (torch.linspace(0.0, 0.3, num_bins), true) },
+                    { opType.TranslateX, (torch.linspace(0.0, 150.0 / 331.0 * image_size.height, num_bins), true) },
```

Comment:
@NiklasGustafsson I think I spotted a bug here, TranslateX should not be image_size.height, it should be image_size.width, TranslateY vice versa.

Reply:
Fixed as part of #1077